### PR TITLE
Fix usage of uninitialized shift vectors with transport

### PR DESCRIPTION
**Pull request by aradi**
Fixes uninitialized usage of shift vectors in transport calculations.

### CODE REVIEW

**Comment by awvwgk** [comment]
We should get the CI for transport working, see #395.

File: prog/dftb+/lib_dftb/scc.F90 [hunk 1]
```diff
@@ -387,6 +390,8 @@ subroutine Scc_initialize(this, env, inp)
       this%h5Correction = inp%h5Correction
     end if
 
+    this%hasExternalShifts = .false.
```
Comment (awvwgk) [inline prog/dftb+/lib_dftb/scc.F90]: Shouldn't we get this information from the input? Setting this here to a fixed value and flipping it later without a way to reset it is somewhat hacky.

File: prog/dftb+/lib_dftb/scc.F90 [hunk 2]
```diff
@@ -875,7 +890,8 @@ subroutine setShiftPerAtom(this, shift)
 
     @:ASSERT(this%tInitialised)
     @:ASSERT(size(shift) == size(this%shiftPerAtom,dim=1))
-    
+
+    this%hasExternalShifts = .true.
```
Comment (awvwgk) [inline prog/dftb+/lib_dftb/scc.F90]: I would rather use something like `@:ASSERT(this%hasExternalShifts)` than setting the variable here.

File: prog/dftb+/lib_dftb/scc.F90 [hunk 3]
```diff
@@ -894,6 +910,7 @@ subroutine setShiftPerL(this, shift)
     @:ASSERT(size(shift,dim=1) == size(this%shiftPerL,dim=1))
     @:ASSERT(size(shift,dim=2) == size(this%shiftPerL,dim=2))
     
+    this%hasExternalShifts = .true.
```
Comment (awvwgk) [inline prog/dftb+/lib_dftb/scc.F90]: Same here. It should be a requirement for setting the shift rather than a consequence.

**Comment by bhourahine** [comment]
> This fix looks a bit hacky. Can't we determine from the input if there will be a shift set?

For the moment, getting the transport working again is the urgent thing. A follow up PR can clean up the logic.

**Pull request merged**